pkg/domain/infra/abi/images.go

```diff
@@ -858,14 +858,22 @@ func execSu(execUser *user.User, command []string) error {
 	return cmd.Run()
 }
 
-func execMachine(execUser *user.User, command []string, machinectl string) error {
-	verb := machinectl
-	args := []string{"shell", "-q", execUser.Username + "@.host"}
+func execSystemd(execUser *user.User, command []string, systemd string) error {
+	cmdLogin, err := utils.LoginUser(execUser.Username)
+	if err != nil {
+		return err
+	}
+	verb := systemd
+	args := []string{"--pty", "-q", "--wait", "--uid", execUser.Uid}
 	if execUser.Uid == "0" {
 		args = append([]string{verb}, args...)
 		verb = "sudo"
 	}
 	cmd := utils.CreateSCPCommand(exec.Command(verb, args...), command)
-	logrus.Debugf("Executing via machinectl: %q", cmd)
-	return cmd.Run()
+	logrus.Debugf("Executing via systemd: %q", cmd)
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+	return cmdLogin.Process.Kill()
```

Comment:
Could you do this as a `defer`, so a failure in lines 874-876 don't leave a stray process?